File: QA-project/backend/src/db/schema/ai-documents.ts
import {
  pgTable,
  serial,
  integer,
  text,
  varchar,
} from "drizzle-orm/pg-core";
import { relations } from "drizzle-orm";
import { datasets } from "./datasets";

// Lightweight reference table in Supabase.
// Actual embedding vectors are stored in Pinecone.
// `pineconeVectorId` links this row to the Pinecone record.
export const aiDocuments = pgTable("ai_documents", {
  docId: serial("doc_id").primaryKey(),
  datasetId: integer("dataset_id")
    .notNull()
    .references(() => datasets.datasetId, { onDelete: "cascade" }),
  chunkText: text("chunk_text").notNull(),
  chunkIndex: integer("chunk_index").notNull(),
  pineconeVectorId: varchar("pinecone_vector_id", { length: 255 }),
});

export const aiDocumentsRelations = relations(aiDocuments, ({ one }) => ({
  dataset: one(datasets, {
    fields: [aiDocuments.datasetId],
    references: [datasets.datasetId],
  }),
}));

File: QA-project/backend/app/models/ai_documents.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base


class AIDocument(Base):
    """Lightweight reference table in Supabase.
    Actual embedding vectors are stored in Pinecone.
    `pinecone_vector_id` links this row to the Pinecone record.
    """
    __tablename__ = "ai_documents"

    doc_id = Column(Integer, primary_key=True, autoincrement=True)
    dataset_id = Column(Integer, ForeignKey("datasets.dataset_id", ondelete="CASCADE"), nullable=False)
    chunk_text = Column(Text, nullable=False)
    chunk_index = Column(Integer, nullable=False)
    pinecone_vector_id = Column(String(255))

    # Relationships
    dataset = relationship("Dataset", back_populates="ai_documents")
